app/src/main/java/com/android/shelfLife/ui/overview/AddFoodItem.kt

```diff
@@ -0,0 +1,217 @@
+package com.android.shelfLife.ui.overview
+
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.android.shelfLife.model.foodFacts.FoodCategory
+import com.android.shelfLife.model.foodFacts.FoodFacts
+import com.android.shelfLife.model.foodFacts.FoodUnit
+import com.android.shelfLife.model.foodFacts.Quantity
+import com.android.shelfLife.model.foodItem.FoodItem
+import com.android.shelfLife.model.foodItem.FoodStatus
+import com.android.shelfLife.model.foodItem.FoodStorageLocation
+import com.android.shelfLife.model.foodItem.ListFoodItemsViewModel
+import com.android.shelfLife.model.household.HouseholdViewModel
+import com.android.shelfLife.ui.navigation.NavigationActions
+import com.google.firebase.Timestamp
+import java.text.SimpleDateFormat
+import java.util.*
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun AddFoodItemScreen(
+    navigationActions: NavigationActions,
+    houseHoldViewModel: HouseholdViewModel,
+    foodItemViewModel: ListFoodItemsViewModel
+) {
+
+  var foodName by remember { mutableStateOf("") }
+  var amount by remember { mutableStateOf("") }
+  var unit by remember { mutableStateOf(FoodUnit.GRAM) }
+  var category by remember { mutableStateOf(FoodCategory.OTHER) }
+  var location by remember { mutableStateOf(FoodStorageLocation.PANTRY) }
+  var expireDate by remember { mutableStateOf("") }
+  var openDate by remember { mutableStateOf("") }
+  var buyDate by remember { mutableStateOf(formatTimestampToDate(Timestamp.now())) }
+
+  var unitExpanded by remember { mutableStateOf(false) }
+  var categoryExpanded by remember { mutableStateOf(false) }
+  var locationExpanded by remember { mutableStateOf(false) }
+
+  Scaffold(
+      modifier = Modifier.fillMaxSize(),
+      topBar = { TopAppBar(title = { Text("Add Food Item") }) },
+  ) { padding ->
```

Comment:
This composable function is currently handling too much in one place, which can make it difficult to maintain and scale over time. I suggest breaking it down into smaller, more focused composable functions, each with a clear, single responsibility. This will improve readability and make the code easier to manage and modify in the future. Ideally, each function should be responsible for one specific task or UI component.